lib/live_view_native/bindings.ex

```diff
@@ -0,0 +1,67 @@
+defmodule LiveViewNative.Bindings do
+  defmacro native_binding(name, type, default) do
+    quote bind_quoted: [name: name, type: type, default: default] do
+      Module.register_attribute(__MODULE__, :_native_bindings, accumulate: true, persist: true)
+
+      @_native_bindings {name, type, default}
+      def _native_bindings, do: @_native_bindings
+      defoverridable [{:_native_bindings, 0}]
+
+      unless Module.get_attribute(__MODULE__, :_defined_native_binding_functions, false) do
+        @_defined_native_binding_functions true
+
+        on_mount {__MODULE__, :_set_native_binding_defaults}
+
+        def on_mount(:_set_native_binding_defaults, _params, _session, socket) do
+          assigns = Enum.map(_native_bindings(), fn {name, _type, default} -> {name, default} end)
+
+          event_payload =
+            Map.new(_native_bindings(), fn {name, _type, default} ->
+              {name, LiveViewNative.JSONCoercable.to_json(default)}
+            end)
+
+          {
+            :cont,
+            socket
+            |> assign(assigns)
+            |> push_event("_live_bindings", event_payload)
```

Comment:
@shadowfacts Do we need to update the `_live_bindings` naming here and within the Swift client as well? I know it's just a socket message and not end-user facing but it would be nice to avoid any confusion with [official LiveView nomenclature](https://hexdocs.pm/phoenix_live_view/bindings.html).

Reply:
Yeah, I'll change the name to "_native_bindings".